v2/internal/util/randextensions/uuid5.go

```diff
@@ -13,27 +13,38 @@ import (
 // Namespace is the ASOv2 UUIDv5 namespace UUID
 var Namespace = uuid.Must(uuid.Parse("9a329043-7ad7-4b1c-812f-9c7a93d6392a"))
 
-// MakeUniqueResourceString generates a string that uniquely identifies a cluster resource.
-func MakeUniqueResourceString(group string, kind string, namespace string, name string) string {
-	return fmt.Sprintf("%s/%s:%s/%s", group, kind, namespace, name)
+// MakeUniqueResourceScopedString generates a string that uniquely identifies a cluster resource. It includes the
+// following distinguishing parts:
+// * Group
+// * Kind
+// * Namespace
+// * Name
+func MakeUniqueResourceScopedString(gk schema.GroupKind, namespace string, name string) string {
```

Comment:
Not clear to me if we want to just not consider the owner ID in the case of RoleDefinition or not... the more I think about it the more I think maybe we should be considering owner here too.